db/consistency_level.cc

```diff
@@ -23,8 +24,64 @@
 #include "locator/abstract_replication_strategy.hh"
 #include "locator/network_topology_strategy.hh"
 #include "heat_load_balance.hh"
+#include "tools/build_info.hh"
 
 namespace db {
+namespace {
+
+// Check the effective replication map consistency:
+// we have an inconsistent effective replication map in case we have a data center
+// with the number of replica nodes higher than it's replication factor.
+void validate_replica_dc_replication_factor(const locator::effective_replication_map& erm,
+                                            const inet_address_vector_replica_set& replicas,
+                                            const consistency_level cl) {
+    // Skip for release builds.
+    if constexpr (tools::build_info::is_release_build()) {
+        return;
+    }
+
+    // Applies only for Local Quorum consistency level.
+    if (cl != consistency_level::LOCAL_QUORUM) {
+        return;
+    }
+
+    const auto& topology = erm.get_topology();
+    const auto& replication_strategy = erm.get_replication_strategy();
+    // Applies only to Network Topology Replication Strategy.
+    if (replication_strategy.get_type() != locator::replication_strategy_type::network_topology) {
+        return;
+    }
+
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-static-cast-downcast)
```

Comment:
Do we have a linter that fires in CI because of this? If not, we shouldn't include such comments in the code.

Reply:
Ok.